src/main/java/dev/codesupport/web/api/service/ArticleServiceImpl.java

```diff
@@ -0,0 +1,225 @@
+        } else {
+             throw new ResourceNotFoundException(ResourceNotFoundException.Reason.NOT_FOUND);
+        }
+        return null;
+    }
+
+    PublishedArticleEntity updatePublishedArticle(PublishedArticleEntity paEntity, ArticleEntity article) {
+        paEntity.setArticleId(article.getId());
+        return paEntity;
+    }
+
+    PublishedArticleEntity createPublishedArticle(ArticleEntity article) {
+        PublishedArticleEntity newPAEntity = new PublishedArticleEntity();
+        newPAEntity.setArticleId(article.getId());
+        newPAEntity.setArticleCode(article.getArticleCode());
+        newPAEntity.setPublished(true);
+        return newPAEntity;
+    }
+
+    static class CrudLogic {
+
+        private final ArticleRepository articleRepository;
+        private final TagRepository tagRepository;
+        private final TagSetRepository tagSetRepository;
+        private final TagSetToTagsRepository tagSetToTagsRepository;
+
+        private CrudLogic(
+                ArticleRepository articleRepository,
+                TagRepository tagRepository,
+                TagSetRepository tagSetRepository,
+                TagSetToTagsRepository tagSetToTagsRepository
+        ) {
+            this.articleRepository = articleRepository;
+            this.tagRepository = tagRepository;
+            this.tagSetRepository = tagSetRepository;
+            this.tagSetToTagsRepository = tagSetToTagsRepository;
+        }
+
+        Consumer<ArticleEntity> preSaveLogic() {
+            return (ArticleEntity articleEntity) -> {
+                updateTagReferences(articleEntity);
+                updateArticleMetaData(articleEntity);
+            };
+        }
+
+        Consumer<ArticleEntity> preGetLogic() {
+            return (ArticleEntity articleEntity) -> {
+                Optional<TagSetEntity> optional = tagSetRepository.findById(articleEntity.getTagSetId());
+                if (optional.isPresent()) {
+                    TagSetEntity tagSet = optional.get();
+                    List<TagSetToTagEntity> xMaps = tagSetToTagsRepository.findAllByTagSetIdOrderByTagId(tagSet.getId());
+                    List<TagEntity> tags = xMaps.stream().map(TagSetToTagEntity::getTag).collect(Collectors.toList());
+                    tagSet.setTags(tags);
+                    articleEntity.setTagSet(tagSet);
+                } else {
+                    log.error("Article (ID: " + articleEntity.getId() + ") had no valid TagSet (ID: " + articleEntity.getTagSetId() + ")");
+                }
+            };
+        }
+
+        void updateArticleMetaData(ArticleEntity articleEntity) {
+            Long articleId = articleEntity.getId();
+            if (articleId != null && articleId != 0) {
+                updateExistingArticle(articleEntity);
+            } else {
+                updateNewArticle(articleEntity);
+            }
+        }
+
+        void updateExistingArticle(ArticleEntity articleEntity) {
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/codesupport/website-backend/pullRequest?prid=6300789)